exporters/prometheus/src/main/java/io/opentelemetry/exporter/prometheus/PrometheusHttpServerBuilder.java

```diff
@@ -18,13 +19,26 @@ public final class PrometheusHttpServerBuilder {
 
   static final int DEFAULT_PORT = 9464;
   private static final String DEFAULT_HOST = "0.0.0.0";
+  private static final MemoryMode DEFAULT_MEMORY_MODE = MemoryMode.IMMUTABLE_DATA;
 
   private String host = DEFAULT_HOST;
   private int port = DEFAULT_PORT;
   private PrometheusRegistry prometheusRegistry = new PrometheusRegistry();
   private boolean otelScopeEnabled = true;
   @Nullable private Predicate<String> allowedResourceAttributesFilter;
   @Nullable private ExecutorService executor;
+  private MemoryMode memoryMode = DEFAULT_MEMORY_MODE;
+
+  PrometheusHttpServerBuilder() {}
+
+  PrometheusHttpServerBuilder(PrometheusHttpServerBuilder builder) {
+    this.host = builder.host;
+    this.port = builder.port;
+    this.prometheusRegistry = builder.prometheusRegistry;
+    this.otelScopeEnabled = builder.otelScopeEnabled;
+    this.allowedResourceAttributesFilter = builder.allowedResourceAttributesFilter;
+    this.executor = builder.executor;
```

Comment:
Don't you want to copy the memory mode as well?

Reply:
Good point